index.js

```diff
@@ -144,132 +127,158 @@ class Disk {
 	}
 	
 	discard(offset, length, callback) {
-		console.log('UNIMPLEMENTED: discarding', length, 'bytes at offset', offset);  // eslint-disable-line no-console
+		this._insertDiskChunk(new diskchunk.DiscardDiskChunk(offset, length));
 		callback(null);
 	}
 
 	getCapacity(callback) {
-		this._getCapacity(callback);
+		if (this.capacity !== null) {
+			callback(null, this.capacity);
+			return;
+		}
+		const self = this;
+		this._getCapacity(function(err, capacity) {
```

Comment:
NIT: I see mixed use of `const self = this` and fat arrow function. We should just use fat arrows

Reply:
This will be fixed in the forthcoming TypeScript PR.